ui/tests/integration/components/sync/secrets/page/overview-test.js

```diff
@@ -72,6 +74,42 @@ module('Integration | Component | sync | Page::Overview', function (hooks) {
     assert.dom(cta.summary).exists('CTA renders');
   });
 
+  test('it should render opt-in modal for enterprise', async function (assert) {
```

Comment:
It seems a little redundant to have two tests clicking through and checking functionality. Is there a reason these were separated into acceptance + component tests? 

It seems reasonable to consolidate these tests, and move these assertions above. This component test would be a great place to assert the adapter error, since that's simpler to do here by just passing in the args